extensions/amp-analytics/0.1/requests.js

```diff
@@ -318,6 +345,73 @@ export class RequestHandler {
     }
     return s.join('&');
   }
+
+  /**
+   * Handle batchInterval
+   */
+  initBatchInterval_() {
+    if (!this.batchInterval_) {
+      return;
+    }
+
+    if (this.maxDelay_) {
+      // Ignore maxDelay in presence of batchInterval
+      // TODO: we can remove the restriction upon requests
+      user().error(TAG,
+          'maxDelay value will not be respected with batchInterval defined');
+      this.maxDelay_ = 0;
+    }
+
+    if (isArray(this.batchInterval_)) {
+      // Support batchInterval in array with absolute time to send request
+      const batchInterval = [];
+      for (let i = 0; i < this.batchInterval_.length; i++) {
+        const current = this.batchInterval_[i];
+        const prev = (i > 0) ? this.batchInterval_[i - 1] : 0;
+        user().assert(isFiniteNumber(current) && Number(current) >= 0,
+            `Invalid batchInterval value: ${this.batchInterval_}, ` +
+            'interval must be number greater than 0.');
+        const interval = (current - prev) * 1000;
+        user().assert(i == 0 || interval > BATCH_INTERVAL_MIN,
+            `Invalid batchInterval value: ${this.batchInterval_}, ` +
+            `interval must be greater than ${BATCH_INTERVAL_MIN / 1000}s.`);
+        batchInterval[i] = interval;
+      }
+      this.batchInterval_ = batchInterval;
+      this.batchIntervalPointer_ = 0;
+    } else if (isFiniteNumber(Number(this.batchInterval_))) {
+      // Support batchInterval in number, send batched requests every interval
+      this.batchInterval_ = Number(this.batchInterval_) * 1000;
+      user().assert(this.batchInterval_ > BATCH_INTERVAL_MIN,
+          `Invalid batchInterval value: ${this.batchInterval_}, ` +
+          `interval must be greater than ${BATCH_INTERVAL_MIN / 1000}s.`);
+    } else {
+      user().assert(false,
+          `Invalid batchInterval value: ${this.batchInterval_}`);
+    }
+
+    this.refreshBatchInterval_();
+  }
+
+  /**
+   * Schedule sending request regarding to batchInterval
+   */
+  refreshBatchInterval_() {
+    let interval;
+    if (this.batchIntervalPointer_ != null) {
+      if (this.batchIntervalPointer_ >= this.batchInterval_.length) {
+        return;
```

Comment:
This would leave execution before resetting the batch interval below. When the end of the interval array is reached, we want to "stick" to the end, so the last element is essentially infinitely repeating.

Reply:
I see. Would there be cases that you want to stop sending requests after a certain amount of time? Like what `batchInterval` is currently doing?

Reply:
Yes, we have the concept of 'reportingWindow' which puts a stop on all reporting (except those with the fire-immediately flag which ignores batching). Typical richmedia config would look like:

{
  batchInterval: [10, 30, 30, 60, 90, 120], // 120 forever... until 20 minutes has passed.
  reportingWindow: 20 // assume units = minutes
}

Expected behavior would be that after the first couple minutes and you reach the 120s value, batches would fire every 120s for perpetuity until 20 minutes have passed.

Reply:
would `batchInterval: [10, 30, 30, 60, 90, 120, 240, ......  1200]` work?

Reply:
Haha technically yes assuming we have the "sticky" logic applied... but that's a lot of 120s.